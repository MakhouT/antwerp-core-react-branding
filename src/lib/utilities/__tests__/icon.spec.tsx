import * as React from "react";

import * as A from "../..";
import { renderAndMatchSnapshot } from "../../../__tests__/__fixtures__/testUtils";

describe("Background", () => {
    it("Renders default", () => {
        // Arrange
        const element = <A.Icon
            name="check"
        />;

        // Act and assert
        const tree = renderAndMatchSnapshot(element);
        expect(tree.props.className).toContain("fa-check");
    });

    it("Renders icon as span", () => {
        // Arrange
        const element = <A.Icon
            name="check"
            span
        />;

        // Act and assert
        renderAndMatchSnapshot(element);
    });

    it("Renders icon with extra classname", () => {
        // Arrange
        const expected = "blabla";
        const element = <A.Icon
            name="check"
            className={expected}
        />;

        // Act and assert
        const tree = renderAndMatchSnapshot(element);
        expect(tree.props.className).toContain(expected);
    });

    it("Renders icon with onclick", () => {
        // Arrange
        const element = <A.Icon
            name="check"
            onClick={() => {/**/ }}
        />;

        // Act and assert
        renderAndMatchSnapshot(element);
    });

});
