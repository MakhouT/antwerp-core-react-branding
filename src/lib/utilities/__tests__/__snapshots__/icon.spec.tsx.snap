// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background Renders default 1`] = `
<i
  className="fa fa-check"
  onClick={[Function]}
  style={undefined}
/>
`;

exports[`Background Renders icon as span 1`] = `
<span
  className="fa fa-check"
  onClick={[Function]}
  style={undefined}
/>
`;

exports[`Background Renders icon with extra classname 1`] = `
<i
  className="fa fa-check blabla"
  onClick={[Function]}
  style={undefined}
/>
`;

exports[`Background Renders icon with onclick 1`] = `
<i
  className="fa fa-check"
  onClick={[Function]}
  style={undefined}
/>
`;
